checks/agentVersionCheck.go

```diff
@@ -0,0 +1,54 @@
+package checks
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/newrelic/newrelic-lambda-extension/config"
+	"github.com/newrelic/newrelic-lambda-extension/lambda/extension/api"
+	"github.com/newrelic/newrelic-lambda-extension/util"
+)
+
+type LayerAgentVersion struct {
+	Version string `json:"version"`
+}
+
+func agentVersionCheck(conf *config.Configuration, reg *api.RegistrationResponse, r runtimeConfig) error {
+	if r.AgentVersion == "" {
+		return nil
+	}
+
+	for i := range r.layerAgentPaths {
+		f := fmt.Sprintf("%s/%s", r.layerAgentPaths[i], r.agentVersionFile)
+		if util.PathExists(f) {
+			b, err := ioutil.ReadFile(f)
+			if err != nil {
+				return nil
+			}
+			v := LayerAgentVersion{}
+			if r.language == Python {
+				v.Version = string(b)
+			} else {
+				_ = json.Unmarshal([]byte(b), &v)
+			}
+			return r.semverValidation(v.Version)
+		}
+	}
+	return nil
+}
+
+func (r runtimeConfig) semverValidation(v string) error {
```

Comment:
This should probably use `semver` module int he stdlib.

https://godoc.org/golang.org/x/mod/semver

Reply:
I was attempting to avoid adding additional packages but if this is preferred I'll replace